contributors/design-proposals/propagation.md

```diff
@@ -142,13 +164,38 @@ distros.
 1. (From @euank) Changing those mountflags may make docker even less stable,
 this may lock up kernel accidentally or potentially leak mounts.
 
+1. (From @jsafrane) Typical container that needs to mount something needs to
+see host's `/dev` and `/sys` as HostPath volumes. This would make them shared
+without any way to opt-out. Docker creates a new `/dev/shm` in the
+container, which gets propagated to the host, shadowing host's `/dev/shm`.
+Similarly, systemd running in a container is very picky about `/sys/fs/cgroup`
+and something prevents it from starting if `/sys` is shared.
 
 ## Decision
 
-We will take 'Make HostPath shared for privileged containers, slave for
-non-privileged', an environment check and an WARNING log will be emitted about
-whether propagation mode is supported.
-
+* We will take 'Add an option in HostPathVolumeSource API'
+  * With an alpha feature gate in 1.8.
+  * Only privileged containers can use `rshared` mount propagation.
+    * When non-privileged container uses `rshared` HostPath, it silently
```

Comment:
The modivation for the silent downgrade is to cover the case where you have a Pod with a shared hostpath volume shared by 2 containers, 1 privileged that does the mounting, and 1 non-privileged that simply reads from the mounted volume. Since we decided to move the mount propagation setting into the volume definition (instead of the volume mount). I don't like making the behavior conditional on privileged either, but I think I prefer it to having the setting in the VolumeMount (since it only applies to a small subset of volumes).